skiko/src/awtMain/cpp/windows/OpenGLLibrary.cc

```diff
@@ -24,62 +24,60 @@ class OpenGL32Library {
-        wglGetProcAddress = (PROC_wglGetProcAddress) GetProcAddress(lib, "wglGetProcAddress");
-        wglMakeCurrent = (PROC_wglMakeCurrent) GetProcAddress(lib, "wglMakeCurrent");
-        wglGetCurrentContext = (PROC_wglGetCurrentContext) GetProcAddress(lib, "wglGetCurrentContext");
+    void load() {
+        if (!isLoaded) {
+            isLoaded = true;
+            lib = LoadLibrary("opengl32.dll");
+            if (lib) {
+                glFinish = (PROC_glFinish) GetProcAddress(lib, "glFinish");
+                glGetIntegerv = (PROC_glGetIntegerv) GetProcAddress(lib, "glGetIntegerv");
+                glGetString = (PROC_glGetString) GetProcAddress(lib, "glGetString");
+                wglCreateContext = (PROC_wglCreateContext) GetProcAddress(lib, "wglCreateContext");
+                wglDeleteContext = (PROC_wglDeleteContext) GetProcAddress(lib, "wglDeleteContext");
+                wglGetProcAddress = (PROC_wglGetProcAddress) GetProcAddress(lib, "wglGetProcAddress");
+                wglMakeCurrent = (PROC_wglMakeCurrent) GetProcAddress(lib, "wglMakeCurrent");
+                wglGetCurrentContext = (PROC_wglGetCurrentContext) GetProcAddress(lib, "wglGetCurrentContext");
+            }
+        }
     }
 };
 
-OpenGL32Library* openGL32Library = nullptr;
-
 extern "C"
 {
     void glFinish(void) {
-        openGL32Library->glFinish();
+        OpenGL32Library::glFinish();
     }
 
     void glGetIntegerv(GLenum pname, GLint *data) {
-        openGL32Library->glGetIntegerv(pname, data);
+        OpenGL32Library::glGetIntegerv(pname, data);
     }
 
     const GLubyte * glGetString(GLenum name) {
-        return openGL32Library->glGetString(name);
+        return OpenGL32Library::glGetString(name);
     }
 
     HGLRC WINAPI wglCreateContext(HDC hDc) {
-        return openGL32Library->wglCreateContext(hDc);
+        return OpenGL32Library::wglCreateContext(hDc);
     }
 
     BOOL WINAPI wglDeleteContext(HGLRC oldContext) {
-        return openGL32Library->wglDeleteContext(oldContext);
+        return OpenGL32Library::wglDeleteContext(oldContext);
     }
 
     PROC WINAPI wglGetProcAddress(LPCSTR lpszProc) {
-        return openGL32Library->wglGetProcAddress(lpszProc);
+        return OpenGL32Library::wglGetProcAddress(lpszProc);
     }
 
     BOOL WINAPI wglMakeCurrent(HDC hDc, HGLRC newContext) {
-        return openGL32Library->wglMakeCurrent(hDc, newContext);
+        return OpenGL32Library::wglMakeCurrent(hDc, newContext);
     }
 
     HGLRC WINAPI wglGetCurrentContext() {
-        return openGL32Library->wglGetCurrentContext();
+        return OpenGL32Library::wglGetCurrentContext();
     }
 
     JNIEXPORT jboolean JNICALL Java_org_jetbrains_skiko_OpenGLLibraryKt_loadOpenGLLibraryWindows(JNIEnv *env, jobject obj) {
-        if (openGL32Library == 0) {
-            HINSTANCE lib = LoadLibrary("opengl32.dll");
-            if (lib) {
-                openGL32Library = new OpenGL32Library(lib);
```

Comment:
No dynamic alloc 🎉  That's better, thanks

Reply:
Not sure that it was matter, as it should only be called once and create a small object